lib/Server.js

```diff
@@ -1738,6 +1735,27 @@ class Server {
 
     await this.initialize();
 
+    let sockets = [];
+
+    this.server.on("connection", (socket) => {
+      // add socket to list
+      sockets.push(socket);
+
+      socket.once("close", () => {
+        // remove socket from list
+        sockets.splice(sockets.indexOf(socket), 1);
+      });
+    });
+
+    this.server.kill = (cb) => {
+      this.server.close(cb);
+      sockets.forEach((socket) => {
+        socket.destroy();
+      });
+      // reset so the server can be restarted
+      sockets = [];
+    };
```

Comment:
Move this to close, we don't need this here

Reply:
Here? https://github.com/webpack/webpack-dev-server/blob/ae8c523854700d63baca335312ccc55df4bfbc42/lib/Server.js#L1790